File: target-sum/target_sum.js
function targetSum(a, t) {
    const dict = {};
    for(let i=0; i< a.length; i++) {
        dict[a[i]] = i;

        const target = t - a[i];
        if(target in dict) {
            return [i, dict[target]];
        }
    }

    return [-1, -1];
}

const a = [1, 2, 5, 6, 7, 9, 11];
const t = 9;
const ans = targetSum(a,t);
console.log("(1,4) => ", ans);

File: target-sum/target_sum.go
package main

import "fmt"

func main() {
	fmt.Println("target sum")

	a := []int{1,2,5,6,7,9,11}
	t := 9

	i, j := targetSum(a, t)
	fmt.Printf("(1,4) => %v, %v\n", i, j) 
}

func targetSum(a []int, t int) (int, int) {
	d := make(map[int]int)
	for i, e := range a {
		d[e] = i

		target := t - a[i]
		if j, ok := d[target]; ok {
			return i, j
		}
	}

	return -1, -1
}
